### DISCUSSION TITLE
changing the combobox with pre selected item.

### DISCUSSION THREAD

**Discussion by aratn** [twilio-labs/paste#2699] [General] [answered]
I have item { item1, item2, item3 } in the combobox and while displaying I want item2 to be preselected. User should be able to change the combobox item to item3 by clicking and selecting the item3. 

Issue: when I am setting value=item2 and onChange={()=>{ /* update the variable to item 3*/}}
Any change is setting it back to item2. user cannot update the combobox value if preselected.

example:
Combobox : once value is set, I cannot change it from the dropdown. https://paste.twilio.design/components/combobox/ 
```
const authors = ['James Baldwin', 'Adrienne Maree Brown', 'Octavia Butler', 'Ta-Nehisi Coates', 'Audre Lorde', 'Nnedi Okorafor'];

const BasicCombobox = () => {
  return (
    <Combobox items={authors} labelText="Select an author" value='Adrienne Maree Brown' required />
  );
};

render(
  <BasicCombobox />
)
```

Input
`<Input aria-describedby="email_help_text" id="email_address" name="email_address" type="email" placeholder="example@twilio.com" value="abc@twilio.com" onChange={(e)=>{console.log(e.target.value);}} required/>`

**Comment by ghost** [comment, marked as answer]
Hi @aratn. Thanks for the discussion topic here. What I'm noticing in your example snippet is that you're using a `value` prop to set the default and a `onChange` prop to manage the controlled component updates. These APIs aren't supported by Combobox within Paste. In fact, if you are using typescript in your project our published type annotations will alert you of this issue at development time prior to executing the code:

![Screen Shot 2022-09-28 at 3 39 39 PM](https://user-images.githubusercontent.com/109178184/192890522-daf24ed3-3230-4f0e-8bf8-603487bc65a6.png)

As for a solution, if you would prefer to allow the DOM to manage the updates (uncontrolled) you can provide a default with a `initialSelectedItem="Adrienne Maree Brown"` prop. If you would like manage the updates explicitly (controlled) you'll want to use a `initialValue="Adrienne Maree Brown"` prop and a `onInputValueChange={() => {}}` prop. We're working to improve the documentation around these subtleties soon. Checkout the component [docs, available API options and examples of controlled component use](https://paste.twilio.design/components/combobox/) for even more context
